src/main/resources/com/puppycrawl/tools/checkstyle/checks/coding/messages.properties

```diff
@@ -66,3 +66,4 @@ unnecessary.semicolon=Unnecessary semicolon.
 unused.local.var=Unused local variable ''{0}''.
 variable.declaration.usage.distance=Distance between variable ''{0}'' declaration and its first usage is {1}, but allowed {2}.
 variable.declaration.usage.distance.extend=Distance between variable ''{0}'' declaration and its first usage is {1}, but allowed {2}.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).
+when.should.be.used=When should be used instead of a single if statement inside a pattern match body.
```

Comment:
```suggestion
when.should.be.used='when' expression should be used instead of a single 'if' statement in switch rule or case statement.
```

Reply:
@rnveach thoughts?

Reply:
I am ok with the general change. I am not sure if there is commonality with other check messages on if we make them technical or keep them in plain English.

Reply:
I have few comments here
> 'when' expression 

is this term is worded like this somewhere ? why we have this extra "expression"

>  instead of a single 'if' statement in switch rule or case statement.

this is not always true because the case label must be a pattern to use `when`. So we may have switch rule and single if statement and we shouldn't suggest using `when`

Reply:
> 'when' expression
is this term is worded like this somewhere ?

https://docs.oracle.com/javase/specs/jls/se22/html/jls-14.html#jls-14.11.1
>  A case label with case patterns may have an optional when expression, known as a guard

I thought the same thing when I saw this. :)

Reply:
>this is not always true because the case label must be a pattern to use when. So we may have switch rule and single if statement and we shouldn't suggest using when

hopefully it is always true for code we are placing violations on :)

Reply:
How about `'when' expression should be used instead of a single 'if' statement in switch block.`?

Reply:
good I think you mean case block 
'when' expression should be used instead of a single 'if' statement in case block.

Reply:
> this is not always true because the case label must be a pattern to use when. So we may have switch rule and single if statement and we shouldn't suggest using when

but still, this isn't satisfied. Do you think it is ok to not mention this explicit fact in the message that `when` can be used with patterns only ?

Reply:
Yes, let’s not mention this in the message.

Reply:
done